pkg/apis/core/validation/validation.go

```diff
@@ -3738,6 +3738,51 @@ func validatePodIPs(pod *core.Pod) field.ErrorList {
 	return allErrs
 }
 
+// validateHostIPs validates IPs in pod status
+func validateHostIPs(pod *core.Pod) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	hostIPsField := field.NewPath("status", "hostIPs")
+
+	// all HostPs must be valid IPs
+	for i, hostIP := range pod.Status.HostIPs {
+		for _, msg := range validation.IsValidIP(hostIP.IP) {
+			allErrs = append(allErrs, field.Invalid(hostIPsField.Index(i), hostIP.IP, msg))
+		}
+	}
+
+	// if we have more than one Pod.PodIP then
+	// - validate for dual stack
+	// - validate for duplication
+	if len(pod.Status.HostIPs) > 1 {
+		hostIPs := make([]string, 0, len(pod.Status.HostIPs))
+		for _, hostIP := range pod.Status.HostIPs {
+			hostIPs = append(hostIPs, hostIP.IP)
+		}
+
+		dualStack, err := netutils.IsDualStackIPStrings(hostIPs)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(hostIPsField, fmt.Errorf("failed to check for dual stack with error:%v", err)))
+		}
+
+		// We only support one from each IP family (i.e. max two IPs in this list).
+		if !dualStack || len(hostIPs) > 2 {
+			allErrs = append(allErrs, field.Invalid(hostIPsField, pod.Status.HostIPs, "may specify no more than one IP for each IP family"))
+		}
+
+		// There should be no duplicates in list of Pod.HostIPs
+		seen := sets.String{}
+		for i, hostIP := range pod.Status.HostIPs {
```

Comment:
you create a new slice to get the hostIPs and now you parse the pod.Status.HostIPs, do we need the allocation or should we use the allocated slice here?

Reply:
This is just validation that the HostIPs values are not duplicated.
Sorry did I miss something? I don't quite understand what you mean.

Reply:
Oh, I read the context carefully, I understand what you mean.